src/plugins/profiling/public/services.ts

```diff
@@ -42,8 +43,24 @@ export function getServices(core: CoreStart): Services {
         const query: HttpFetchQuery = {
           index: 'profiling-events',
           projectID: 5,
-          timeFrom: timeFrom,
-          timeTo: timeTo,
+          timeFrom,
+          timeTo,
+          // TODO remove hard-coded value for topN items length and expose it through the UI
+          n: 100,
+        };
+        return await core.http.get(paths.FlamechartElastic, { query });
+      } catch (e) {
+        return e;
+      }
+    },
+
+    fetchElasticFlamechart2: async (timeFrom: number, timeTo: number) => {
```

Comment:
Why duplicating the function? It seems like the only change is the index prefix, can it be passed as a parameter to the existing function? AFAIR we still have it in the query string

Reply:
It is less time to duplicate the code. We will remove it in the end, so why touch the original code here ? Also this allows making experimental changes to the duplicated code and reversion is a no-brainer (just removal of the code).